extra_tests/snippets/stdlib_json.py

```
@@ -239,3 +239,19 @@ class Dict(dict):
     RecursionError,
     lambda: json.loads(('[{"x":' * _deep) + "1" + ("}]" * _deep)),
 )
+
+
+# Invalid \uXXXX escape: error position points at the 'u', matching CPython.
+try:
+    json.loads('"\\uXYZW"')
+except json.JSONDecodeError as e:
+    assert e.pos == 2, f"expected pos=2, got {e.pos}"
+else:
+    assert False, "expected JSONDecodeError"
```

Ruff (0.15.11)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)